### PR TITLE
Implement configuration option for the querier to only hit the ingesters based on the trace id hashes instead of all healthy ingesters

### LINKED ISSUE #1477: Implement configuration option for the querier to only hit the ingesters based on the trace id hashes instead of all healthy ingesters.
**Is your feature request related to a problem? Please describe.**
This is a feature request for Trace by ID API call on Query Frontend. We understand that each call gets sharded and parallel requests are sent to Querier and Ingester to fetch the trace based on the ID, but as per the [code](https://github.com/grafana/tempo/blob/main/modules/querier/querier.go#L194) , are we making a call to all healthy ingesters in the ring. 

**Describe the solution you'd like**
This can lead to issues with fetch, even though one of the ingesters are unhealthy(for RF=1), we will not be able to query the ingesters as it will fail with _Too many unhealthy replicas_ error. We watch to fetch the ingester instance based on the hash of trace id (how we find out the ingester instance during distributor writes by using the filter on [ringstrategy](https://github.com/grafana/tempo/blob/35d7e668d7c8466e372f99edaa99007be5eb2b03/vendor/github.com/grafana/dskit/ring/batch.go#L58)

**Describe alternatives you've considered**
We can increase the RF to increase failure tolerance but that has an associated cost.

cc: @bikashmishra100 , @ashwinidulams

### PR DESCRIPTION
**Pull request by sagarwala**
**What this PR does**:
Implement configuration option for the querier to only hit the ingesters based on the trace id hashes instead of all healthy ingesters

**Which issue(s) this PR fixes**:
Fixes #1477

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: modules/querier/config.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ type Config struct {
 	ExtraQueryDelay         time.Duration `yaml:"extra_query_delay,omitempty"`
 	MaxConcurrentQueries    int           `yaml:"max_concurrent_queries"`
 	Worker                  worker.Config `yaml:"frontend_worker"`
+	QuerySingleIngester     bool          `yaml:"query_single_ingester"`
```
Comment (joe-elliott) [inline modules/querier/config.go:20]: Let's change this parameter name. When using RF3 your change would actually be querying 3 ingesters, not 1.

I'm not coming up with a good name. If you have a good idea, please use it over :).
`query_ingesters_from_ring`? `query_relevant_ingesters`?

File: docs/tempo/website/configuration/_index.md [hunk 2]
```diff
@@ -341,6 +341,12 @@ querier:
     # not distinguish between the types of queries.
     [max_concurrent_queries: <int> | default = 5]
 
+    # The query frontend sents sharded requests to ingesters and querier (/api/traces/<id>)
+    # By default, healthy ingesters are queried for the trace id.
```
Comment (joe-elliott) [inline docs/tempo/website/configuration/_index.md:345]:
```suggestion
    # By default, all healthy ingesters are queried for the trace id.
```

File: docs/tempo/website/configuration/_index.md [hunk 3]
```diff
@@ -341,6 +341,12 @@ querier:
...
+    # This value controls if single ingester is to be retrieved based on the trace id hash.
```
Comment (joe-elliott) [inline docs/tempo/website/configuration/_index.md:346]:
```suggestion
    # When true the querier will hash the trace id in the same way that distributors do.
    # The querier will then only query those ingesters who own the trace id hash as determined by the ring as opposed to querying all ingesters.
```

**Review by joe-elliott — COMMENTED** [review]
Change looks good. Had a few comments and suggestions.

**Comment by sagarwala** [comment]
Thank you @joe-elliott . I have updated the PR as per the comments. I am using `query_relevant_ingesters` for the config name :)

**Review by joe-elliott — APPROVED** [review]
Nice addition. Thanks!

**Pull request merged**